### PR TITLE
:sparkles: - feat: allow custom onClick/disabled behavior in Kanban c…

### PR DESCRIPTION
**Pull request by svenvandescheur**
…omponent items

### CODE REVIEW

File: src/components/data/kanban/kanban.tsx [hunk 1]
```diff
@@ -608,6 +610,11 @@ export const KanbanButton: React.FC<KanbanButtonProps> = ({
     <ButtonLink
       align="start"
       className="mykn-kanban__button"
+      disabled={
+        typeof object.disabled === "undefined"
+          ? !href
+          : Boolean(object.disabled)
+      }
```
Comment (SilviaAmAm) [inline src/components/data/kanban/kanban.tsx:617]: I wonder: wouldn't it be simpler to have a prop "disabled" and let the parent component manage the logic of whether the button is disabled?

File: src/components/typography/p/p.scss [hunk 2]
```diff
@@ -19,4 +19,12 @@
     font-size: var(--typography-font-size-body-xs);
     line-height: var(--typography-line-height-body-xs);
   }
+
+  &--word-break-break-all {
+    word-break: break-all;
+  }
+
+  &--work-break-break-word {
```
Comment (SilviaAmAm) [inline src/components/typography/p/p.scss:27]:
```suggestion
  &--word-break-break-word {
```
:smile:

File: src/components/typography/p/p.scss [hunk 3]
```diff
@@ -19,4 +19,12 @@
...
+    word-break: break-word;
```
Comment (SilviaAmAm) [inline src/components/typography/p/p.scss:28]: I think that this one is deprecated? https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#syntax

File: src/components/typography/p/p.tsx [hunk 4]
```diff
@@ -28,12 +32,14 @@ export const P: React.FC<PProps> = ({
...
   muted = false,
   size = "s",
+  wordBreak,
   ...props
 }) => (
   <p
     className={clsx("mykn-p", `mykn-p--size-${size}`, {
       ["mykn-p--bold"]: bold,
       ["mykn-p--muted"]: muted,
+      [`mykn-p--work-break-${wordBreak}`]: wordBreak,
```
Comment (SilviaAmAm) [inline src/components/typography/p/p.tsx:42]:
```suggestion
      [`mykn-p--word-break-${wordBreak}`]: wordBreak,
```

File: src/components/data/kanban/kanban.tsx (hunk 1 above)
Comment (svenvandescheur) [reply src/components/data/kanban/kanban.tsx:617]: There are a few scenario's where an item's disabled can be set:

- There is a prop (object attribute) set that indicates disabled (`typeof object.disabled !== "undefined"`), in such case the prop should be used.
- There is no prop (object attribute) set that indicates disabled (`typeof object.disabled === "undefined"`), in such cased the kanban should base it's state on whether a `href` or `onClick` is set that provided interaction. 

I don't think the first case is used anywhere, but I can imagine that the automatic behavior (second scenario) might be problematic is certain cases so I think is sensible to provide an escape hatch.

**Review by SilviaAmAm — APPROVED** [review]

**Pull request merged**